chapter02/chapter02-03.adoc

```diff
@@ -0,0 +1,319 @@
+
+=== Creating an Entity class
+
+An Entity class represents a specific object, in our case a product. It contains the product's details id and name, 
+and other properties like price, and quantity. To implement an entity class first, you need to create a `Product`  class, as below:
+
+[source, java]
+----
+// Product.java
+package io.microprofile.tutorial.store.product.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Product {
+   private Long id;
+   private String name;
+   private String description;
+   private Double price;
+}
+----
+
+Explanation: 
+
+* The `Product` class is a Plain Old Java Object (POJO). It has an `id`, `name`, `description` and `price` property. The `id` property is of type `Long`, The `name` and `description` properties are of type `String`. The `price` property is of type `Double`. 
+
+* `@Data` annotation will generate constructors, getters, and setters for all fields. By doing this, 
+you enable the Jackson library to convert your Java objects to JSON and vice versa. All properties must be of `Object` type as well.
+Jackson cannot work with primitive types because they cannot be `null`.
+
+* `@AllArgsConstructor` generates a constructor with one argument for each field in the class. 
+This is useful for instantiating objects with all their fields initialized. 
+
+* `@NoArgsConstructor` generates a default constructor 
+for the class.
+
+=== Creating a Resource class
+
+A resource class represents a collection of related resources. It includes methods for creating, updating, deleting, and retrieving 
+(CRUD) operations on the resources. Let us now create a `ProductResource` class with a `getProducts()` method to return a list of 
+`Product` objects.
+
+[source, java]
+----
+// ProductResource.java
+package io.microprofile.tutorial.store.product.resource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.microprofile.tutorial.store.product.entity.Product;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/products")
+@ApplicationScoped
+public class ProductResource {
+   private List<Product> products;
+
+   public ProductResource() {
+       products = new ArrayList<>();
+
+       products.add(new Product(Long.valueOf(1L), "iPhone", "Apple iPhone 15", Double.valueOf(999.99)));
+       products.add(new Product(Long.valueOf(2L), "MacBook", "Apple MacBook Air", Double.valueOf(1299.99)));
+   }
+
+   @GET
+   @Produces(MediaType.APPLICATION_JSON)
+   public List<Product> getProducts() {
+       // Return a list of products
+       return products;
+   }   
+}
+----
+  
+Explanation: 
+
+* The `ProductResource` is annotated with `@ApplicationScoped`. This will ensure that this class is available as long as the 
```

Comment:
```suggestion
* The `ProductResource` is annotated with `@ApplicationScoped`. This will ensure that only one instance of this class available when the 
```